src/Tests.cmake

```diff
@@ -1,9 +1,380 @@
 include_guard()
 
+include("${CMAKE_CURRENT_LIST_DIR}/SystemLink.cmake")
+
 # Enable coverage reporting for gcc/clang
 function(enable_coverage _project_name)
   if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
     target_compile_options(${_project_name} INTERFACE --coverage -O0 -g)
     target_link_libraries(${_project_name} INTERFACE --coverage)
   endif()
 endfunction()
+
+function(_configure_target target_name type)
```

Comment:
I like how this function lumps together all the CMake functions making things more declarative. Could you name it as something like `add_target`?

Reply:
The problem here is whether I should split `DEPENDENCIES_CONFIG`, `DEPENDENCIES` and `LIBRARIES` to `[PRIVATE|PUBLIC|INTERFACE]_DEPENDENCIES_CONFIG` and so on.

Currently this function only supports `"library_test"` (executable) and `"test_config"` (header-only library), they are expected to only link one dependency type (PRIVATE for executable and INTERFACE for header-only library).

But for a non-header-only library, it might link a PRIVATE library, a PUBLIC library and a INTERFACE library at the same time. If this function is named as `add_blah`, users might expect this function also works on non-header-only libraries, so for a correct linkage, `DEPENDENCIES_CONFIG` must be split to `[PRIVATE|PUBLIC|INTERFACE]_DEPENDENCIES_CONFIG`.